### PR TITLE
Fix deletion of augeas elements

### PR DESCRIPTION
**Pull request by joseivanlopez**
When nil is passed to AugeasTree#delete, all elements are removed. Fix method to control when nil is passed.

https://github.com/config-files-api/config_files_api/pull/13#issuecomment-264861107

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -1,4 +1,10 @@
 -------------------------------------------------------------------
+Mon Dec  5 15:38:36 UTC 2016 - joseivanlopez@gmail.com
+
+- fix deletion of augeas elements
```
Comment (jreidinger) [inline CHANGELOG]: id is needed, you can reuse bsc from previous changelog entry

Comment (jreidinger) [inline CHANGELOG]: and please be more verbose in changes. Maybe something like `fix regression when passing nil to AugeasTree#delete`

Comment (joseivanlopez) [inline CHANGELOG]: Ok

**Comment by jreidinger** [comment]
LGTM

**Pull request merged**